src/project-manager.ts

```diff
@@ -337,6 +338,24 @@ export class ProjectManager implements Disposable {
 		});
 	}
 
+	isConfigDependency(filePath: string): boolean {
+		for (const config of this.configurations()) {
+			config.ensureConfigFile();
+			if (config.isExpectedDeclarationFile(filePath)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	ensureConfigDependencies(): Observable<never> {
+		return observableFromIterable(this.inMemoryFs.uris())
+			.filter(uri => this.isConfigDependency(uri2path(uri)))
+			.mergeMap(uri => this.updater.ensure(uri))
+			.publishReplay()
+			.refCount();
```

Comment:
why publishReplay + refCount?

Reply:
I looked at ensureModuleStructure, which I also understood would be invoked many times but should always return the same (cached) sequence. But I'd love to know how this use case differs (and remove it if unnecessary).

Reply:
this doesn't actually save the (i.e. cache) the Observable, so it doesn't has a use here. If you want to always return the same Observable you need to save it in a property